index.js

```diff
@@ -88,10 +98,12 @@ const oauthPlugin = fp(function (fastify, options, next) {
   }
 
   const cbk = function (o, code, callback) {
-    return callbackify(o.oauth2.getToken.bind(o.oauth2, {
+    const body = Object.assign({}, generateTokenRequestParams(tokenRequestParams), {
```

Comment:
I don't understand this extra call to `generateTokenRequestParams`.
It cannot be set by the user (it is hidden into `credentials.auth[kGenerateTokenRequestParams]`) and the default `defaultGenerateTokenRequestParams` returns the input object itself.

Why it is necessary?

Reply:
I tried to implement the same principle like for `generateAuthorizationUri`. The logic there allows two ways how to set additional callback params:

* by the user via config parameter
* by the plugin maintainer for preconfigured connections (see `oauthPlugin.APPLE_CONFIGURATION`)

As said I tried to follow the same code style like for the callback parameters. If the callback implementation for `APPLE_CONFIGURATION` was a very specific workaround, I agree that this implementation is probably not needed for this PR.

Just let me know if I should simplify it.

Reply:
Yes, please remove it as it is not necessary

Reply:
done